File: yandex_back_prep/leetcode/1446. Consecutive Characters.py
class Solution:
    def maxPower(self, s: str) -> int:
        
        res = ''
        le = -1

        if len(s) == 1:
            return 1

        pos = 0
        while pos < len(s)-1:
            res += s[pos]
            while  pos < len(s)-1 and s[pos] == s[pos+1]:
                pos +=1
                res += s[pos]
        
            le = max(le, len(res))
            res = ''
            pos +=1

        return le

File: yandex_back_prep_cpp/1446. Consecutive Characters.cpp
#include <limits>
#include <algorithm>

class Solution
{
public:
    int maxPower(string s)
    {

        if (s.size() == 1)
        {
            return 1;
        }

        int max_len = std::numeric_limits<int>::lowest();

        int l = 0;
        int r = 0;

        while (r < s.size() - 1)
        {
            while (r < s.size() - 1 && s[r] == s[r + 1])
            {
                r++;
            }
            max_len = std::max(max_len, r - l + 1);
            r++;
            l = r;
        }

        return max_len;
    }
};